README.md

```diff
@@ -1,30 +1,119 @@
-needs.
-
-To use it just hit `Use this template` button or [this link][generate].
-
-Things to do with your newly created repo:
-
-1. Run`devctl replace -i "REPOSITORY_NAME" "$(basename $(git rev-parse
-   --show-toplevel))" --ignore '.git/**' '**'`.
-2. Run `devctl replace -i "template" "$(basename $(git rev-parse
-   --show-toplevel))" --ignore '.git/**' '**'`.
-3. Go to https://github.com/giantswarm/REPOSITORY_NAME/settings and make sure `Allow
-   merge commits` box is unchecked and `Automatically delete head branches` box
-   is checked.
-4. Go to https://github.com/giantswarm/REPOSITORY_NAME/settings/access and add
-   `giantswarm/bots` with `Write` access and `giantswarm/employees` with
-   `Admin` access.
-5. Add this repository to https://github.com/giantswarm/github.
-6. Create quay.io docker repository if needed.
-7. Add the project to the CircleCI:
-   https://circleci.com/setup-project/gh/giantswarm/REPOSITORY_NAME
-8. Change the badge (with style=shield):
-   https://circleci.com/gh/giantswarm/REPOSITORY_NAME.svg?style=shield&circle-token=TOKEN_FOR_PRIVATE_REPO
-   If this is a private repository token with scope `status` will be needed.
-
-[generate]: https://github.com/giantswarm/template/generate
+# helm-values-gen
+
+`helm-values-gen` generates a YAML payload that contains all default values in
+a given JSON schema. This is useful when maintaining a helm project with a
+`values.schema.json` that already contains the default values that are
+expected in `values.yaml`.
+
+## Installation
+
+Currently, `helm-values-gen` can be installed as go binary.
+In the future it will be (additionally) possible to install it as a helm plugin.
+
+```nohighlight
+go install github.com/giantswarm/helm-values-gen@latest
+```
+
+## Usage
+
+Executing `schemalint verify` without any options will check whether a file is
+valid JSON Schema and whether it is normalized.
+
+```nohighlight
+$ helm-values-gen values.schema.json -o values.yaml
+
+Wrote default values to values.yaml
+```
+
+Use `--help` to learn about more options.
+
+## Details
+
+As JSON schemas can be nested deeply, we need to define which default values
+will be contained in the resulting payload.
+The tool follows these steps:
+
+1. Start in the root of the schema
+2. If there is a default value, add it to the payload.
+3. Recurse into all properties.
+
+**Note**:
+The tool does not recurse into the `items` keyword of an array.
+To specify default values for an array you have to set a default on the level
+of the array.
+
+Consider the following example:
+
+values.schema.json
+
+```json
+{
+  "properties": {
+    "addresses": {
+      "items": {
+        "properties": {
+          "city": {
+            "type": "string",
+            "default": "New York"
+          }
+        },
+        "type": "object"
+      },
+      "type": "array"
+    },
+    "lastName": {
+      "type": "string",
+      "default": "Doe"
+    }
+  },
+  "type": "object"
+}
+```
```

Comment:
I suggest to order the `type` key to the top, to improve readability

Reply:
Done